lib/src/config/config.rs

```diff
@@ -119,7 +119,11 @@ impl Config {
                 return Err(self.bad_type(name, val, "an unsigned integer"));
             }
 
-            self.port = port as usize;
+            if port > (u16::max_value() as i64) {
```

Comment:
> Also with the declaration of port if you want it as a u16 do

But then I can't add a test for ever u16::max, and would instead rely on the parse error which I'm thinking is worse?? Also, I couldn't test for the `<0` and would have to rely on the parse error. Hmm, I'm seeing what you are saying though.